android/app/src/main/kotlin/net/defined/mobile_nebula/NebulaVpnService.kt

```diff
@@ -121,6 +121,16 @@ class NebulaVpnService : VpnService() {
             builder.addRoute(unsafeIPNet.network, unsafeIPNet.maskSize.toInt())
         }
 
+        // Add our dns resolvers
+        site!!.dnsResolvers.forEach { dnsResolver ->
+            builder.addDnsServer(dnsResolver)
+        }
+
+        val cm = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        cm.allNetworks.forEach { network ->
+            cm.getLinkProperties(network).dnsServers.forEach { builder.addDnsServer(it) }
```

Comment:
` cm.getLinkProperties(network)?.dnsServers?.forEach { builder.addDnsServer(it) }`

Would not compile otherwise.

Reply:
Come to think of it, lines 129 to 132 should be removed entirely, as they cause ta DNS leak on Android when tested with https://mullvad.net/ (but no others, interestingly). Removing these lines fixes the DNS leak. They are also based on a false assumption that the device will not roam between networks.